llvm/lib/Transforms/InstCombine/InstCombineShifts.cpp

```diff
@@ -428,7 +428,10 @@ Instruction *InstCombinerImpl::commonShiftTransforms(BinaryOperator &I) {
         return R;
 
   Constant *CUI;
-  if (match(Op1, m_ImmConstant(CUI)))
+  if (match(Op1, m_Constant(CUI)) &&
+      (!isa<ConstantExpr>(CUI) ||
```

Comment:
Can you add a fixme here to note that we should remove this special case after we use ConstantInt to represent a scalable vector splat by default?

Reply:
Done in the matcher for a0f4ac6e2ef13f90ce44524a6943f98583e43367